test/lsptoolshost/integrationTests/index.ts

```diff
@@ -3,11 +3,71 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
+import * as fsExtra from 'fs-extra';
+import path from 'path';
+import { AggregatedResult } from '@jest/test-result';
 import { runIntegrationTests } from '../../runIntegrationTests';
 import { jestIntegrationTestProjectName } from './jest.config';
+import { activateCSharpExtension } from './integrationHelpers';
 
 export async function run() {
     process.env.RUNNING_INTEGRATION_TESTS = 'true';
 
-    await runIntegrationTests(jestIntegrationTestProjectName);
+    await activateCSharpExtension();
+    await moveLogs('activated');
+
+    let anyFailures = false;
+    let results: AggregatedResult | undefined;
+    if (process.env.TEST_FILE_FILTER) {
+        results = await runIntegrationTests(jestIntegrationTestProjectName);
+        await moveLogs(path.basename(process.env.TEST_FILE_FILTER, '.integration.test.ts'));
+        anyFailures = anyFailures || !results.success;
+    } else {
+        const workingDirectory = process.cwd();
+
+        const testFiles = (await fsExtra.readdir(__dirname))
+            .filter((file) => file.endsWith('.integration.test.js'))
+            .map((file) => path.join(__dirname, file));
+
+        for (let file of testFiles) {
+            // We have to fix up the file path because the test file was discovered in the /out/ directory.
+            file = file.substring(0, file.length - 2) + 'ts';
+            file = file.replace(/[\\/]out[\\/]/, path.sep);
+            file = workingDirectory[0] + file.substring(1);
```

Comment:
So locally `__dirname` is returning the drive letter lowercased and the filepaths also are lowercased. Jest fails to match the paths to a test suite if I don't bring in a correct drive letter. Thankfully the working directory has the correct casing. Figure on unix-like platforms it should basically be a no-op.